src/Summoner/Source.hs

```diff
@@ -18,26 +18,34 @@ data Source
     = Url Text
     -- | File path to the local source file.
     | File FilePath
+    -- | Link to source source file.
+    | Link Text
     deriving (Show, Eq)
 
 sourceT :: Key -> BiToml Source
 sourceT nm = Toml.match (Toml._Text   >>> _Url)  (nm <> "url")
          <|> Toml.match (Toml._String >>> _File) (nm <> "file")
+         <|> Toml.match (Toml._Text >>> _Link) (nm <> "link")
   where
     _Url :: BiMap Text Source
     _Url = Toml.invert $ Toml.prism (source Just (const Nothing)) Url
 
     _File :: BiMap FilePath Source
     _File = Toml.invert $ Toml.prism (source (const Nothing) Just) File
 
+    _Link :: BiMap Text Source
+    _Link = Toml.invert $ Toml.prism (source Just (const Nothing)) Link
+
     source :: (Text -> c) -> (FilePath -> c) -> Source -> c
```

Comment:
I think that tests are failing because of the fact that you handle different constructors with the same function. Try adding one extra parameter to `source` function and write:

```haskell
source _ _ f (Link x) = f x
```